src/commands/tileindex-validate/tileindex.validate.ts

```diff
@@ -357,6 +362,35 @@ export interface TiffLocation {
   bands: string[];
 }
 
+/**
+ * Reproject the bounding box if the source and target projections are different.
+ * @param bbox input bounding box
+ * @param sourceProjection CRS of the input bounding box
+ * @param targetProjection target CRS
+ */
+function reprojectIfNeeded(bbox: BBox, sourceProjection: Projection, targetProjection: Projection): BBox | null {
+  {
+    if (targetProjection !== sourceProjection) {
+      const [ulX, ulY] = targetProjection.fromWgs84(sourceProjection.toWgs84([bbox[0], bbox[3]]));
+      const [lrX, lrY] = targetProjection.fromWgs84(sourceProjection.toWgs84([bbox[2], bbox[1]]));
+      if (ulX === undefined || ulY === undefined || lrX === undefined || lrY === undefined) {
```

Comment:
```suggestion
      if (ulX == null || ulY == null || lrX == null || lrY == null) {
```

You often run into incorrect types where it says something is undefined, but in reality its null and vice versa.

Reply:
resolved this by asserting that we get `[number,number]`